### PR TITLE
lxd: Download Alpine image instead of relying on external source

### LINKED ISSUE #97484: Find stable urls for lxd container tests
Please find stable urls for the lxd container tests

https://github.com/NixOS/nixpkgs/blob/6aeccf3bcbb02c6f31b56a942017e45d52a87cb0/nixos/tests/lxd.nix#L10

these fail to download now and are causing test failures:

```
trying https://uk.images.linuxcontainers.org/images/alpine/3.11/i386/default/20200608_13:00/lxd.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

trying https://uk.images.linuxcontainers.org/images/alpine/3.11/i386/default/20200608_13:00/rootfs.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
created 5239 symlinks in user environment
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
error: cannot download lxd.tar.xz from any mirror
```

cc @Patryk27

### PR DESCRIPTION
**Pull request by Patryk27**
###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/issues/97484.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Patryk27** [comment]
cc @jonringer

**Comment by worldofpeace** [comment]
Let's not have these files in nixpkgs please. We could have an admin upload these to `https://tarballs.nixos.org`

**Comment by worldofpeace** [comment]
@Patryk27 I contacted @AmineChikhaoui from the infra team to upload these files to tarballs.nixos.org for you.
We should hear back with stable urls and hashes soon to use here :+1:

**Comment by Patryk27** [comment]
Sure, got it - thank you :-)

**Comment by AmineChikhaoui** [comment]
Hey @Patryk27 , can you provide the latest stable url from uk.images.linuxcontainers.org or similar for me to use to download the tarballs first. Thanks

**Comment by Patryk27** [comment]
@AmineChikhaoui hi - those are the files:
- https://uk.images.linuxcontainers.org/images/alpine/3.12/i386/default/20200909_13:00/lxd.tar.xz
- https://uk.images.linuxcontainers.org/images/alpine/3.12/i386/default/20200909_13:00/rootfs.tar.xz

**Comment by AmineChikhaoui** [comment]
Uploaded to the following urls:
```
$ nix-prefetch-url  http://tarballs.nixos.org/alpine/3.12/lxd.tar.xz
path is '/nix/store/cafixfv42cfn63g8bbg592rxkrhnsmp9-lxd.tar.xz'
1c5q9b0cmg04gc66djw51fh160bw3d9yrgv1gvm4nfk5xxl0mmyn

$ nix-prefetch-url  http://tarballs.nixos.org/alpine/3.12/rootfs.tar.xz
path is '/nix/store/z3inz6kkrwj5nzn7d46av7iyfp7j926h-rootfs.tar.xz'
084h3ffl5lmzvkxgxcz0416rgjhcg6xbnfcf5m8cp20s5aqvvdjd
```

**Comment by Patryk27** [comment]
kk, ready - thanks :-)

**Comment by worldofpeace** [comment]
@ofborg test lxd

**Comment by worldofpeace** [comment]
If you could also change the commit to `nixosTests.lxd: ...`  and squash the two commits together than would be great :+1:

**Comment by worldofpeace** [comment]
It seems to fail on ofborg
```
machine: must succeed: mkdir /var/lxd-pool
machine # [   18.636334] lxd[627]: t=2020-09-09T17:36:49+0000 lvl=warn msg=" - Couldn't find the CGroup blkio.weight, I/O weight limits will be ignored"
(0.05 seconds)
machine: must succeed: cat /nix/store/64l16k960yab010xv69spwrqbvmjam72-config.yaml | lxd init --preseed
machine # Error: Failed to connect to local LXD: Get "http://unix.socket/1.0": dial unix /var/lib/lxd/unix.socket: connect: no such file or directory
machine: output:
error:
Traceback (most recent call last):
  File "/nix/store/kcy6z1kbf4qd6jvv6y9c9g43sdsxy466-nixos-test-driver/bin/.nixos-test-driver-wrapped", line 898, in run_tests
    exec(tests, globals())
  File "<string>", line 1, in <module>
  File "<string>", line 10, in <module>
  File "/nix/store/kcy6z1kbf4qd6jvv6y9c9g43sdsxy466-nixos-test-driver/bin/.nixos-test-driver-wrapped", line 421, in succeed
    raise Exception(
Exception: command `cat /nix/store/64l16k960yab010xv69spwrqbvmjam72-config.yaml | lxd init --preseed` failed (exit code 1)
cleaning up
```

**Comment by Patryk27** [comment]
@worldofpeace could you try again? I've added `machine.wait_for_file("/var/lib/lxd/unix.socket")` to ensure that LXD's socket has been actually initialized.

File: nixos/tests/lxd.nix [hunk 1]
```diff
@@ -6,15 +6,14 @@ let
   #
   # I've chosen to import Alpine Linux, because its image is turbo-tiny and,
   # generally, sufficient for our tests.
-
   alpine-meta = pkgs.fetchurl {
-    url = "https://uk.images.linuxcontainers.org/images/alpine/3.11/i386/default/20200608_13:00/lxd.tar.xz";
-    sha256 = "1hkvaj3rr333zmx1759njy435lps33gl4ks8zfm7m4nqvipm26a0";
+    url = "https://tarballs.nixos.org/alpine/3.12/lxd.tar.xz";
+    sha256 = "sha256-1tcKaO9lOkvqfmG/7FMbfAEToAuFy2YMewS8ysBKuLA=";
```
Comment (edolstra) [inline nixos/tests/lxd.nix:11]:
```suggestion
    hash = "sha256-1tcKaO9lOkvqfmG/7FMbfAEToAuFy2YMewS8ysBKuLA=";
```

**Comment by worldofpeace** [comment]
@ofborg test lxd

**Comment by worldofpeace** [comment]
Thanks @Patryk27, could you also backport this to 20.09?

**Comment by Patryk27** [comment]
Sure - how do I do that? Just prepare MR to the `release-20.09` branch?

**Comment by worldofpeace** [comment]
@Patryk27 See https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#backporting-changes

**Pull request merged**